src/Quaternion.jl

```diff
@@ -29,15 +39,54 @@ quat(p, v1, v2, v3, n) = Quaternion(p, v1, v2, v3, n)
 quat(x) = Quaternion(x)
 quat(s, a) = Quaternion(s, a)
 
+"""
+    real(q::Quaternion)
+
+Return the real part of the quaternion `q`.
+
+See also: [`imag_part`](@ref), [`quat`](@ref), [`isreal`](@ref).
+
+# Examples
+```jldoctest
+julia> real(quat(1,2,3,4))
+1
+```
```

Comment:
Like in Base, I think we also want to capture the signature and examples of `real(::Type{<:Quaternion})` and `real(::Array{<:Quaternion})`

Reply:
I added `real([quat(5,6,7,8), 9])` example to the docstring of `real(q::Quaternion)` because we don't define `real(A::AbstractArray)` and cannot add docstring to that.